reference/opt/shaders-msl/desktop-only/comp/extended-arithmetic.desktop.comp

```diff
@@ -0,0 +1,177 @@
+#include <metal_stdlib>
+#include <simd/simd.h>
+
+using namespace metal;
+
+struct SSBOUint
+{
+    uint a;
+    uint b;
+    uint c;
+    uint d;
+    uint2 a2;
+    uint2 b2;
+    uint2 c2;
+    uint2 d2;
+    uint3 a3;
+    uint3 b3;
+    uint3 c3;
+    uint3 d3;
+    uint4 a4;
+    uint4 b4;
+    uint4 c4;
+    uint4 d4;
+};
+
+struct ResType
+{
+    uint _m0;
+    uint _m1;
+};
+
+struct ResType_1
+{
+    uint2 _m0;
+    uint2 _m1;
+};
+
+struct ResType_2
+{
+    uint3 _m0;
+    uint3 _m1;
+};
+
+struct ResType_3
+{
+    uint4 _m0;
+    uint4 _m1;
+};
+
+struct SSBOInt
+{
+    int a;
+    int b;
+    int c;
+    int d;
+    int2 a2;
+    int2 b2;
+    int2 c2;
+    int2 d2;
+    int3 a3;
+    int3 b3;
+    int3 c3;
+    int3 d3;
+    int4 a4;
+    int4 b4;
+    int4 c4;
+    int4 d4;
+};
+
+struct ResType_4
+{
+    int _m0;
+    int _m1;
+};
+
+struct ResType_5
+{
+    int2 _m0;
+    int2 _m1;
+};
+
+struct ResType_6
+{
+    int3 _m0;
+    int3 _m1;
+};
+
+struct ResType_7
+{
+    int4 _m0;
+    int4 _m1;
+};
+
+kernel void main0(device SSBOUint& u [[buffer(0)]], device SSBOInt& i [[buffer(1)]])
+{
+    ResType _25;
+    _25._m0 = u.a + u.b;
+    _25._m1 = select(uint(1), uint(0), addsat(u.a, u.b) == UINT_MAX);
```

Comment:
Sure this is correct? I would expect overflow detection to be `result < max(a, b)`.